document/api/src/main/java/io/camunda/document/api/DocumentStore.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.document.api;
+
+import java.io.InputStream;
+
+public interface DocumentStore {
```

Comment:
If an implementation itself is async, wouldnt it be better to support an async store interface and let the implementation handle it (for example the S3 client library?). It looks like we have a blocking interface and an async client wrapper to call it. Curious about your thoughts on this.

Reply:
Async S3 Client interface: https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/S3AsyncClient.html

Reply:
:+1: I can recommend to check out the backup store API which is very similar. All methods return futures here: https://github.com/camunda/camunda/blob/main/zeebe/backup/src/main/java/io/camunda/zeebe/backup/api/BackupStore.java

Reply:
Good idea, initially I didn't like how the method signature was becoming too long like `CompletableFuture<DocumentOperationResponse<DocumentReference>>` if we used the async store implementation. But I can clearly see the benefit of having an async interface to support async store clients directly. I'll give it another try.